### PR TITLE
move bottom sheet from fragments to activities

### LINKED ISSUE #612: Scrolling in profiles does not pull up the header (in current master build)
So I just found a bug that if you go to a user page and try to scroll through the user's toots without raising the finger, instead of pushing the header image, etc. up, you'll just scroll thorugh that small part in which the toots are shown.  
This is not the case in the 1.7 build.
Here's a screencast of it (sorry for the bad quality 😅):
https://drive.google.com/open?id=1LLD5HkDAAvhCA60-8xKdYgClP0G4rZgb

* * * *
- Tusky Version: Current Master build (3dfe43dfb2b7712ae9eceff18c07b534b9f63d65)
- Android Version: 7.1.2
- Android Device: Moto Z2 Play
- Mastodon instance (if applicable): mastodon.xyz

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### PR DESCRIPTION
**Pull request by connyduck**
closes #612 

seems to work fine and should also be a bit more efficient since there are less bottom sheets and CoordinatorLayouts inflated

### CODE REVIEW

**Comment by connyduck** [comment]
this now also fixes the issue where mentions in account notes did not get openend in Tusky

what do you think?

File: app/src/main/java/com/keylesspalace/tusky/AccountActivity.java [hunk 1]
```diff
@@ -61,6 +61,8 @@
 import com.pkmmte.view.CircularImageView;
 import com.squareup.picasso.Picasso;
 
+import org.jetbrains.annotations.NotNull;
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/AccountActivity.java]: Why not Android support one?

File: app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+    override fun onPostCreate(savedInstanceState: Bundle?) {
+        super.onPostCreate(savedInstanceState)
+
+        val bottomSheetLayout: LinearLayout = findViewById(R.id.item_status_bottom_sheet)
+            bottomSheet = BottomSheetBehavior.from(bottomSheetLayout)
+            bottomSheet.state = BottomSheetBehavior.STATE_HIDDEN
+            bottomSheet.setBottomSheetCallback(object : BottomSheetBehavior.BottomSheetCallback() {
+                override fun onStateChanged(bottomSheet: View, newState: Int) {
+                    when (newState) {
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt]: when -> if ?

File: app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                        }
+                    }
+                }
+
+                override fun onSlide(bottomSheet: View, slideOffset: Float) {}
+            })
+
+    }
+
+    open fun viewUrl(url: String) {
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt]: Why methods are open? If it's for mocking, we use 'mockito-inline' which doesn't care

File: app/src/test/java/com/keylesspalace/tusky/BottomSheetActivityTest.kt [hunk 4]
```diff
@@ -283,29 +283,32 @@ class SFragmentTest {
...
+    class FakeBottomSheetActivity(val api: MastodonApi) : BottomSheetActivity() {
+
         var status: Status? = null
         var accountId: String? = null
-        var url: String? = null
+        var link: String? = null
 
         init {
-            callList = mutableListOf()
+            bottomSheet = Mockito.mock(BottomSheetBehavior::class.java) as BottomSheetBehavior<LinearLayout>
```
Comment (charlag) [inline app/src/test/java/com/keylesspalace/tusky/BottomSheetActivityTest.kt]: I think we need to suppress this

**Comment by charlag** [comment]
Also tests are failing because of that bottom sheet... I wonder if suppress will help

File: app/src/main/java/com/keylesspalace/tusky/AccountActivity.java (hunk 1 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/AccountActivity.java]: sry probably autocomplete

File: app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt (hunk 3 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt]: how can I do this without complicating the code a lot?

File: app/src/main/java/com/keylesspalace/tusky/BaseActivity.java [hunk 5]
```diff
@@ -37,12 +37,17 @@
...
 
+import javax.inject.Inject;
+
 import retrofit2.Call;
 
 public abstract class BaseActivity extends AppCompatActivity {
 
     protected List<Call> callList;
 
+    @Inject
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/BaseActivity.java]: Huh? Does it work like that? I thought you can inject only in concrete classes?

File: app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt (hunk 3 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt]: Wait, I don't understand. You made them open for mocking, right?
You just do `when(activity.methodCall()).doReturn(restult)`. Nothing changes with them being open or not.
I feel like I don't understand something.

Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt]: no, the class is not mocked mith Mockito, but subclassed as FakeBottomSheetActivity

Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/BottomSheetActivity.kt]: Ah, I see now, it's a poor man's Spy. We'll fix it sometime

**Pull request merged**